src/backends/pytorch/torch_backend.cc

```diff
@@ -0,0 +1,687 @@
+        "' to Torch datatype");
+  }
+
+  torch::Tensor input_tensor = torch::from_blob(content, shape, pr.second.code, device_);
+
+  if ((input_tensor.numel() * pr.second.bits / 8) != byte_size) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "unexpected size " + std::to_string(byte_size) +
+        " for inference input '" + name + "', expecting " +
+        std::to_string(input_tensor.nbytes()));
+  }
+  inputs_.push_back(input_tensor);
+
+  return Status::Success();
+}
+
+Status
+LibTorchBackend::Context::ReadFixedSizedOutputTensor(
+    const std::string& name, const DataType dtype,
+    const size_t dtype_byte_size, const size_t total_batch_size,
+    std::vector<Scheduler::Payload>* payloads)
+{
+  std::vector<int64_t> content_shape;
+  const char* content = nullptr;
+  size_t byte_size = 0;
+  RETURN_IF_ERROR(GetOutputTensor(name, dtype, &content, &byte_size,
+      &content_shape));
+
+  const size_t total_byte_size =
+      GetElementCount(content_shape) * dtype_byte_size;
+  const size_t batch1_byte_size = total_byte_size / total_batch_size;
+
+  if (byte_size != total_byte_size) {
+    return Status(
+        RequestStatusCode::INTERNAL,
+        "unexpected size for output '" + name + "', byte-size " +
+            std::to_string(byte_size) + " does not equal " +
+            std::to_string(total_batch_size) + " * " +
+            std::to_string(batch1_byte_size));
+  }
+
+  size_t content_offset = 0;
+
+  for (auto& payload : *payloads) {
+    const InferRequestHeader& request_header =
+        payload.request_provider_->RequestHeader();
+    const size_t expected_byte_size =
+        request_header.batch_size() * batch1_byte_size;
+
+    // If 'payload' requested this output then copy it from
+    // 'content'. If it did not request this output then just
+    // skip it in the 'content'.
+    if ((payload.response_provider_ != nullptr) &&
+        payload.response_provider_->RequiresOutput(name)) {
+      void* buffer;
+      Status status = payload.response_provider_->AllocateOutputBuffer(
+          name, &buffer, expected_byte_size, content_shape);
+      if (status.IsOk()) {
+        memcpy(buffer, content + content_offset, expected_byte_size);
+      }
+
+      if (!status.IsOk()) {
+        payload.status_ = status;
+      }
+    }
+
+    content_offset += expected_byte_size;
+  }
+
+  return Status::Success;
+}
+
+Status
+LibTorchBackend::Context::GetOutputTensor(
+    const std::string& name, const DataType dtype,
+    const char** content, size_t* byte_size,
+    std::vector<int64_t>* content_shape)
+{
+  // Initialize char* content[output_flat.nbytes()];
+  torch::DeviceType output_device = torch::kCPU;
+  try{
+    outputs_ = outputs_.to(output_device)
+    torch::Tensor output_flat = outputs_.flatten();
+    std::vector<float> outputs_vector;
+    for(int i=0;i<output_flat.sizes()[0];i++){
+      outputs_vector.push_back(output_flat[i].item().to<float>());
```

Comment:
Seems to be an extra copy, now you are doing "`output_flat` -> `outputs_vector` -> `content`". You should be able to copy from `output_flat` to `content` directly

Reply:
Resolved to copy directly from `output_flat` -> `content`.